src/wallet/rpcdump.cpp

```diff
@@ -616,9 +616,19 @@ UniValue dumpwallet(const JSONRPCRequest& request)
 
     EnsureWalletIsUnlocked(pwallet);
 
-    std::ofstream file;
     boost::filesystem::path filepath = request.params[0].get_str();
     filepath = boost::filesystem::absolute(filepath);
+
+    /* Prevent arbitrary files from being overwritten. There have been reports
+     * that users have overwritten wallet files this way:
+     * https://github.com/bitcoin/bitcoin/issues/9934
+     * It may also avoid other security issues.
+     */
+    if (boost::filesystem::exists(filepath)) {
```

Comment:
What if a `dumpwallet` is executed after this check with the same filename, or the filename is created by other means? IMO we should use `fopen` with `wx` mode where `x` stands for:

> File access mode flag "x" can optionally be appended to "w" or "w+" specifiers. This flag forces the function to fail if the file exists, instead of overwriting it. (C11)

and drop `exists()` test.

Reply:
I'm not terribly worried about race conditions here, just about overwriting existing files such as the wallet.

Reply:
Yes, the comment should have paranoia label..